### PR TITLE
feat: initial support for dark mode

### LINKED ISSUE #141: Implement dark mode + light mod for the website
I would love to try and do this

### PR DESCRIPTION
**Pull request by xhayper**
closes #141

TODO:
- [ ] Polish some color up
- [ ] Fix transition
- [ ] Improve the theme code

### CODE REVIEW

File: web/info/src/app.html [hunk 1]
```diff
@@ -4,6 +4,22 @@
...
     <meta name="viewport" content="width=device-width" />
+
+    <script>
+      // Prevent FOUC
+      if (
+        localStorage.theme === 'dark' ||
+        (!('theme' in localStorage) &&
+          window.matchMedia('(prefers-color-scheme: dark)').matches)
+      ) {
+        document.documentElement.classList.add('dark');
```
Comment (kiosion) [inline web/info/src/app.html:15]: This should be done thru `<svelte:head use:classList={...} />`, JS shouldn't be used inside the app.html template - https://svelte.dev/tutorial/svelte-head

File: web/info/src/components/nav/theme-button.svelte [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<script lang="ts">
+  import { Menu, MenuButton } from '@rgossiaux/svelte-headlessui';
+  import { APP_THEMES } from '$lib/constants';
+  import themeWritable from '$stores/theme';
+
+  let theme: number = APP_THEMES.light;
+
+  themeWritable.subscribe((value) => {
+    theme = value;
+  });
```
Comment (kiosion) [inline web/info/src/components/nav/theme-button.svelte:10]: You can use the auto-subscription syntax of `$theme` here instead - https://svelte.dev/tutorial/auto-subscriptions

Comment (xhayper) [reply web/info/src/components/nav/theme-button.svelte:10]: Thank you so much, this is my first time using svelte!

File: web/info/src/app.html (hunk 1 above)
Comment (xhayper) [reply web/info/src/app.html:15]: Thanks!

**Comment by xhayper** [comment]
#143 is the better version of this, so i will be closing it

**Pull request closed without merging**